### PR TITLE
Almost finished main web page with non-working search. Functionality …

### PR DESCRIPTION
**Pull request by ArinaSavchenko**
…is not implemented at all (+3 squashed commit)

Squashed commit:

[67a1d02] Top of the web page

[188a077] Top of the web page

[10d9635] Top of the web page (+1 squashed commits)

Squashed commits:

[69a396f] Top of the web page

(cherry picked from commit 0adb16463d74c5a25f8e3715a6dadd167680f21d)

### CODE REVIEW

File: ClientApp/src/app/app.module.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import { MatAutocompleteModule } from '@angular/material/autocomplete';
+import { MatRadioModule } from '@angular/material/radio';
+import { MatDatepickerModule } from '@angular/material/datepicker';
+import { MatNativeDateModule} from '@angular/material/core';
+
+import { AppComponent } from './app.component';
+import { TopBarComponent } from './top-bar/top-bar.component';
+import { MainPageComponent } from './main-page/main-page.component';
+import { CarouselComponent } from './carousel/carousel.component';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
```
Comment (bagatka) [inline ClientApp/src/app/app.module.ts]: BrowserAnimationsModule in **components** import :(

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.css [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+:host ::ng-deep .mat-radio-inner-circle{
+  height: 12px;
+  width: 12px;
+}
+:host ::ng-deep .mat-radio-button .mat-radio-ripple{
+  height: 20px; /*double of your required circle radius*/
+  width: 20px;  /*double of your required circle radius*/
+  left: calc(50% - 10px); /*'10px'-same as your required circle radius*/
+  top: calc(50% - 10px); /*'10px'-same as your required circle radius*/
```
Comment (bagatka) [inline ClientApp/src/app/flights-search-bar/flights-search-bar.component.css]: Comments are not allowed on master

File: ClientApp/src/app/top-bar/top-bar.component.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  selector: 'app-top-bar',
+  templateUrl: './top-bar.component.html',
+  styleUrls: ['./top-bar.component.css']
+})
+export class TopBarComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (bagatka) [inline ClientApp/src/app/top-bar/top-bar.component.ts]: Do you need an empty constructor and an empty method?

File: ClientApp/src/app/app.module.ts (hunk 1 above)
Comment (ArinaSavchenko) [reply ClientApp/src/app/app.module.ts:19]: I'm about to block you

Comment (ArinaSavchenko) [reply ClientApp/src/app/app.module.ts]: > BrowserAnimationsModule in **components** import :(

Sasha, I'm about to block you

Comment (andrei-kruhliakou-itechart) [reply ClientApp/src/app/app.module.ts]: But he's actually right :) Please use logical grouping for your imports.

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.css (hunk 2 above)
Comment (andrei-kruhliakou-itechart) [reply ClientApp/src/app/flights-search-bar/flights-search-bar.component.css]: Correction - obvious or meaningless comments shouldn't be left in the code. Same goes for commented out pieces of code. 
In this particular case not sure if information here adds any value to the code but I leave it to you to decide.

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+@Component({
+  selector: 'app-flights-search-bar',
+  templateUrl: './flights-search-bar.component.html',
+  styleUrls: ['./flights-search-bar.component.css']
+})
+export class FlightsSearchBarComponent implements OnInit{
+  ticketsType: number;
+  myControl = new FormControl();
+  options: string[] = ['Minsk', 'Moscow', 'Milan', 'Rome', 'Stockholm', 'Helsinki'];
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flights-search-bar/flights-search-bar.component.ts:14]: Consider moving fake data to separate service which will be replaced with the real one fetching data from the backend later.
This change isn't required for current PR.

File: ClientApp/src/app/flights/flights.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<p>flights works!</p>
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flights/flights.component.html]: Please do not include non-functional and unfinished components in pull requests.

File: ClientApp/src/app/main-page/main-page.component.css [hunk 6]
```diff
@@ -0,0 +1,15 @@
+.container {
+  position: relative;
+}
+.search-bar, .image_carousel{
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/main-page/main-page.component.css:5]: Please separate blocks in css with empty lines.

File: ClientApp/src/app/search-bar/search-bar.component.css [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  background-color: white;
+  border: solid darkgray;
+}
+
+.links{
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+.mat-tab-label{
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/search-bar/search-bar.component.css:14]: Empty line here, please

File: ClientApp/src/app/search-bar/search-bar.component.html [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<div class="search">
+  <mat-tab-group class="links" dynamicHeight>
+    <mat-tab label="Flights">
+      <app-flights-search-bar></app-flights-search-bar>
+    </mat-tab>
+    <mat-tab label="Online check-in">
+      Content 2
+    </mat-tab>
+    <mat-tab label="My bookings">
+      Content 3
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/search-bar/search-bar.component.html]: Please use some meaningful text here or ho text at all

File: ClientApp/src/app/search-bar/search-bar.component.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './search-bar.component.html',
+  styleUrls: ['./search-bar.component.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class SearchBarComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/search-bar/search-bar.component.ts:14]: Please do not leave empty methods unless they are required for code to work.

File: ClientApp/src/app/top-bar/top-bar.component.html [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+  <a routerLink="/airline" title="Go to main page">
+    <img id="logo" src="assets/S7_new_logo.svg.jpg" alt="Logo" width="150" >
+  </a>
+
+  <div id="account_buttons">
+
+    <a id="log-in" mat-flat-button routerLink="/log-in">Log in</a>
+    <a id="sign-up" mat-flat-button routerLink="/sign-up">Sign up</a>
+
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/top-bar/top-bar.component.html]: Unnecessary empty lines

File: ClientApp/src/index.html [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  <meta charset="utf-8">
+  <title>Airline</title>
+  <base href="/">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="icon" type="image/x-icon" href="assets/airplane2.ico">
+  <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500&display=swap" rel="stylesheet">
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+</head>
+<body class="mat-typography">
+<app-root></app-root>
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/index.html]: Missing indent

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: ClientApp/src/app/app.module.ts [hunk 12]
```diff
@@ -1,8 +1,10 @@
 import { BrowserModule } from '@angular/platform-browser';
-import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { NgModule } from '@angular/core';
+
+import { AppRoutingModule } from './app-routing.module';
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/app.module.ts]: This should be moved lower to separate framework and libraries imports from application code imports.

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

**Pull request merged**